src/main/java/org/traccar/database/NotificationManager.java

```diff
@@ -92,9 +92,10 @@ public void updateEvent(Event event, Position position) {
                     if (getById(notificationId).getType().equals(event.getType())) {
                         boolean filter = false;
                         if (event.getType().equals(Event.TYPE_ALARM)) {
-                            String alarms = notification.getString("alarms");
-                            if (alarms == null || !alarms.contains(event.getString(Position.KEY_ALARM))) {
```

Comment:
What happened to the null check?

Reply:
The null check is still there `if (alarmsAttribute != null)` if non null get the array of alarm types. Since split will never return null, we are safe

Reply:
Yeah, but before the logic was to set `filter = true` when the string is null. You removed that logic and now if it's null, it will always be false. Am I missing something?

Reply:
You are right, we need to check when it is `null` and then set `filter=true`

Reply:
fixed